src/neighbors.cc

```diff
@@ -571,7 +524,7 @@ neighbors_init(void)
                 debugs(15, DBG_IMPORTANT, "WARNING: Peer looks like this host." <<
                        Debug::Extra << "Ignoring cache_peer " << *thisPeer);
 
-                neighborRemove(thisPeer);
+                NeighborRemove(thisPeer.get());
```

Comment:
Does removing items from the being-iterated vector constitute undefined behavior?